src/routes/transactions/transactions.service.ts

```diff
@@ -70,4 +73,52 @@ export class TransactionsService {
       results,
     };
   }
+
+  async getModuleTransactions(
+    chainId: string,
+    routeUrl: Readonly<URL>,
+    safeAddress: string,
+    to?: string,
+    module?: string,
+    paginationData?: PaginationData,
+  ): Promise<Page<ModuleTransaction>> {
+    const domainTransactions = await this.safeRepository.getModuleTransactions(
+      chainId,
+      safeAddress,
+      to,
+      module,
+      paginationData?.limit,
+      paginationData?.offset,
+    );
+
+    const safeInfo = await this.safeRepository.getSafe(chainId, safeAddress);
+
+    const results = await Promise.all(
+      domainTransactions.results.map(
+        async (domainTransaction) =>
+          new ModuleTransaction(
+            await this.moduleTransactionMapper.mapTransaction(
+              chainId,
+              domainTransaction,
+              safeInfo,
+            ),
+          ),
+      ),
+    );
+    const nextURL = cursorUrlFromLimitAndOffset(
+      routeUrl,
+      domainTransactions.next,
+    );
+    const previousURL = cursorUrlFromLimitAndOffset(
+      routeUrl,
+      domainTransactions.previous,
+    );
+    const result = <Page<ModuleTransaction>>{
+      next: nextURL?.toString(),
+      previous: previousURL?.toString(),
```

Comment:
We're including explicit nulls when the next/previous values are not present, see https://github.com/5afe/safe-client-gateway-nest/pull/208/files 🙂

Reply:
I'll rebase when the PR is merged :)

Reply:
Rebased :)